src/types.rs

```diff
@@ -619,6 +621,50 @@ impl From<AcmeRenewalInfoCertIdentifier<'_>> for EncodedCertificateIdentifier {
     }
 }
 
+#[cfg(feature = "x509-parser")]
+impl<'a> TryFrom<&'a [u8]> for EncodedCertificateIdentifier {
```

Comment:
I think I'd prefer to skip this impl in favor of only the one for `&CertificateDer`? IIRC the conversion from `&[u8]` to `CertificateDer` is ~free anyway, and that would give us some more (explicit) type safety.

Reply:
Done.